metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/lineage.py

```diff
@@ -517,19 +517,24 @@ def lineage_via_catalog_lineage_api(
                     ]
                 )
 
-            # Convert project tables to <project_id>.<dataset_id>.<table_id> format
-            project_table_names = list(
-                map(
-                    lambda table: "{}.{}.{}".format(
-                        table.project, table.dataset_id, table.table_id
-                    ),
-                    project_tables,
-                )
-            )
-
             lineage_map: Dict[str, Set[LineageEdge]] = {}
             curr_date = datetime.now()
-            for table in project_table_names:
+            for project_table in project_tables:
+
+                if not is_schema_allowed(
+                    self.config.dataset_pattern,
+                    schema_name=project_table.dataset_id,
+                    db_name=project_table.project,
+                    match_fully_qualified_schema_name=self.config.match_fully_qualified_names,
+                ) or not self.config.table_pattern.allowed(project_table.table_id):
```

Comment:
`self.config.table_pattern` needs to be checked against qualified table name `<project>.<dataset>.<table>` . I would suggest moving the table variable definition 
` table = f"{project_table.project}.{project_table.dataset_id}.{project_table.table_id}"` 
before if condition and using it here.

Reply:
Thank you, I didn't notice that table patterns expected qualified names. I'll have to update my own recipes as well. Updated in 5cbb1b1